tests/closure.rs

```diff
@@ -0,0 +1,390 @@
+        .unwrap()
+        .register_taskset(taskset)
+        .execute()
+        .unwrap();
+
+    // Ensure that the right paths have been called.
+    for (idx, item) in test_endpoints.iter().enumerate() {
+        let mock_endpoint = &mock_endpoints[idx];
+
+        let format_item = |message, assert_item| {
+            return format!("{} for item = {:#?}", message, assert_item);
+        };
+
+        // Confirm that we loaded the mock endpoint.
+        assert!(
+            mock_endpoint.times_called() > 0,
+            format_item("Endpoint was not called > 0", &item)
+        );
+        let expect_error = format_item("Item does not exist in goose_metrics", &item);
+        let endpoint_metrics = goose_metrics
+            .requests
+            .get(&format!("GET {}", item.path))
+            .expect(&expect_error);
+
+        assert!(
+            endpoint_metrics.path == item.path,
+            format_item(
+                &format!("{} != {}", endpoint_metrics.path, item.path),
+                &item
+            )
+        );
+        assert!(endpoint_metrics.method == GooseMethod::GET);
+
+        // Confirm that Goose and the server saw the same number of page loads.
+        let status_code: u16 = item.status_code;
+
+        assert!(
+            endpoint_metrics.response_time_counter == mock_endpoint.times_called(),
+            format_item("response_time_counter != times_called()", &item)
+        );
+        assert!(
+            endpoint_metrics.status_code_counts[&status_code] == mock_endpoint.times_called(),
+            format_item("status_code_counts != times_called()", &item)
+        );
+        assert!(
+            endpoint_metrics.success_count == mock_endpoint.times_called(),
+            format_item("success_count != times_called()", &item)
+        );
+        assert!(
+            endpoint_metrics.fail_count == 0,
+            format_item("fail_count != 0", &item)
+        );
+    }
+
+    // Test specific things directly access the mock endpoints here.
+    let index = &mock_endpoints[0];
+    let about = &mock_endpoints[1];
+
+    // Confirm that we loaded the index roughly three times as much as the about page.
+    let one_third_index = index.times_called() / 3;
+    let difference = about.times_called() as i32 - one_third_index as i32;
+    assert!(difference >= -2 && difference <= 2);
+
+    // Verify that Goose started the correct number of users.
+    assert!(goose_metrics.users == config.users.unwrap());
+}
+
+#[test]
+#[cfg_attr(not(feature = "gaggle"), ignore)]
+// Spawn a gaggle of 1 manager and 2 workers each simulating one user. Run a load test,
+// with a single task set containing two weighted tasks setup via closure. Validate
+// that weighting and metrics are correctly merged to the Manager.
+fn test_single_taskset_closure_gaggle() {
+    use std::sync::Arc;
+
+    // Configure endpoints to test.
+    let test_endpoints = vec![
```

Comment:
Uh - no ;) -- we really should not be duplicating the code here, but instead call a common helper function.

Especially as we re-organize it here, this is perfect to add the helper function. (can first be in this file)

Reply:
My main concern is currently all integration tests are excellent tutorials. The more we abstract them, the less useful they become for this purpose. In any case, I do understand we make life easier for ourselves as we abstract.

Reply:
@LionsAd I've reworked all tests affected by this PR, moving duplicated code into helper functions.